kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/config/NamedFilterDefinition.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config;
+
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.kroxylicious.proxy.filter.FilterFactory;
+import io.kroxylicious.proxy.plugin.PluginImplConfig;
+import io.kroxylicious.proxy.plugin.PluginImplName;
+
+/**
+ * A named filter definition
+ * @param name The name of the filter instance.
+ * @param type
+ * @param config
+ * @see Configuration#filterDefinitions()
+ */
+public record NamedFilterDefinition(
+                                    @JsonProperty(required = true) String name,
+                                    @PluginImplName(FilterFactory.class) @JsonProperty(required = true) String type,
+                                    @PluginImplConfig(implNameProperty = "type") Object config) {
+
+    @JsonCreator
+    public NamedFilterDefinition {
+        Objects.requireNonNull(name);
+        // TODO should probably constrain the allowd chars in the name
```

Comment:
To support the `Filter` kind in the operator we want the allowed syntax to allow RFC1123 domain names (see https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names), but it could be less restrictive if we have a reason.

But without a reason, I'm inclined to constrain filter names to follow exactly the requirements of RFC1123 domain names for now. We can always loosen it later.

This also makes me think: We should consider _some_ rule for `virtualClusters`.